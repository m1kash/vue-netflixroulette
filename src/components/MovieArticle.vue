<template>
  <div class="flex">
    <div class="w-1/4 pr-4">
      <img :src="poster_path" alt="image" width="270">
    </div>
    <div class="w-3/4 space-y-4 font-extralight">
      <h1 class="text-4xl text-zinc-200 after:inline-flex after:justify-center after:items-center after:border-zinc-500 after:border after:border-solid after:rounded-full after:ml-5 after:text-3xl after:w-16 after:h-16 after:text-green-500 after:content-[attr(data-rating)]" :data-rating="vote_average">{{ title }}</h1>
      <div class="text-xl text-zinc-300">{{ allGenres }}</div>
      <div class="text-rose-500 text-2xl">{{ year }} &nbsp; {{ duration }}</div>
      <p class="text-xl text-zinc-300"> {{ overview }}</p>
    </div>
  </div>
</template>

<script lang="ts">
import {defineComponent, PropType} from "vue";

export default defineComponent({
  name: 'movie-article',
  props: {
    title: {
      required: true,
      type: String,
    },
    poster_path: {
      required: true,
      type: String,
    },
    genres: {
      required: true,
      type: Array as PropType<string[]>,
    },
    id: Number,
    runtime: {
      required: true,
      type: Number,
    },
    vote_average: {
      required: true,
      type: Number,
    },
    overview: String,
    release_date: {
      required: true,
      type: String,
    },
  },
  setup: () => ({
  }),
  computed: {
    allGenres(): string {
      return this.$props.genres.join('&');
    },
    year(): string {
      return this.$props.release_date.split('-')[0];
    },
    duration(): string {
      return `${this.$props.runtime} min`;
    }
  }
})
</script>

<style scoped>

</style>
