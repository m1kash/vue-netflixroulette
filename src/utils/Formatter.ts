import IFormatter from "@/types/IFormatter";

const Formatter: IFormatter = {
  year(date) {
    const DateInstance = new Date(date) as Date;

    return DateInstance.getFullYear();
  },
  separator(items, separator) {
    return items.join(separator);
  },
  duration(min) {

    return `${min || 0} min`;
  }
};

export default Formatter;
