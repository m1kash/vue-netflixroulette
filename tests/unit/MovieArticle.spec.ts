import 'jest';
import { shallowMount, mount, VueWrapper } from '@vue/test-utils';
import MovieArticle from '@/components/MovieArticle.vue';
import data from "../mock-data/movies.json";
import Formatter from "@/utils/Formatter";

describe('MovieArticle.vue', () => {
  let wrapper: VueWrapper;
  let filterSearchMock: jest.Mock;

  beforeEach(() => {
    filterSearchMock = jest.fn();
    wrapper = mount(MovieArticle, {
      props: {
        ...data[0]
      },
      global: {
        provide: {
          'formatter': Formatter
        },
      }
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });

  test('render correct', () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
