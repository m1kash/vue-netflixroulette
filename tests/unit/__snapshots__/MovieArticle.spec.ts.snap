// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieArticle.vue render correct 1`] = `
<div
  budget="0"
  class="flex"
  revenue="0"
  tagline=""
  vote_count="9"
>
  <div
    class="w-1/4 pr-4"
  >
    <img
      alt="image"
      src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
      width="270"
    />
  </div>
  <div
    class="w-3/4 space-y-4 font-extralight"
  >
    <h1
      class="text-4xl text-zinc-200 after:inline-flex after:justify-center after:items-center after:border-zinc-500 after:border after:border-solid after:rounded-full after:ml-5 after:text-3xl after:w-16 after:h-16 after:text-green-500 after:content-[attr(data-rating)]"
      data-rating="2"
    >
      Test 1
    </h1>
    <div
      class="text-xl text-zinc-300"
    >
      Action Adventure Science Fiction
    </div>
    <div
      class="text-rose-500 text-2xl"
    >
      2020 Â  0 min
    </div>
    <p
      class="text-xl text-zinc-300"
    >
      The third film based on Marvel's Guardians of the Galaxy.
    </p>
  </div>
</div>
`;
